extension/jmxmetricextension/integration_test.go

```diff
@@ -0,0 +1,358 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build integration
+
+package jmxmetricextension
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"runtime"
+	"testing"
+	"time"
+
+	prommodel "github.com/prometheus/common/model"
+	promconfig "github.com/prometheus/prometheus/config"
+	promdiscovery "github.com/prometheus/prometheus/discovery"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config/confignet"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/exporter/exportertest"
+	"go.opentelemetry.io/collector/receiver/otlpreceiver"
+	"go.opentelemetry.io/collector/receiver/prometheusreceiver"
+	"go.opentelemetry.io/collector/testbed/testbed"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest/observer"
+)
+
+type JmxIntegrationSuite struct {
+	suite.Suite
+	JarPath string
+}
+
+func TestJmxIntegration(t *testing.T) {
+	suite.Run(t, new(JmxIntegrationSuite))
+}
+
+func (suite *JmxIntegrationSuite) SetupSuite() {
+	jarPath, err := downloadJmxMetricGathererJar()
+	require.NoError(suite.T(), err)
+	suite.JarPath = jarPath
+}
+
+func (suite *JmxIntegrationSuite) TearDownSuite() {
+	require.NoError(suite.T(), os.Remove(suite.JarPath))
+}
+
+func downloadJmxMetricGathererJar() (string, error) {
+	url := "https://oss.jfrog.org/artifactory/list/oss-snapshot-local/io/opentelemetry/contrib/opentelemetry-java-contrib-jmx-metrics/0.0.1-SNAPSHOT/opentelemetry-java-contrib-jmx-metrics-0.0.1-20200918.184353-3.jar"
+	resp, err := http.Get(url)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	file, err := ioutil.TempFile("", "jmx-metrics.jar")
+	if err != nil {
+		return "", err
+	}
+
+	defer file.Close()
+	_, err = io.Copy(file, resp.Body)
+	return file.Name(), err
+}
+
+func cassandraContainer(t *testing.T) testcontainers.Container {
+	ctx := context.Background()
+	req := testcontainers.ContainerRequest{
+		FromDockerfile: testcontainers.FromDockerfile{
+			Context:    path.Join(".", "testdata"),
+			Dockerfile: "Dockerfile.cassandra",
```

Comment:
Is this because Cassandra exposes a JMX port? Isn't there a more lightweight example? :-)

Reply:
Good point.  Will update with a basic jre version and query a system mbean.